### DISCUSSION TITLE
Add custom query key?

### DISCUSSION THREAD

**Discussion by Yuhao-C** [trpc/trpc#4023] [Q&A]
Sometimes a query may implicitly depend on some `tRPC` context (such as `session`) which is not directly used as a query parameter. If I would like to make the query depend on the `session` (so that if user status changed, it would refetch automatically), with `react-query`, I could simply add it to the `queryKey`. For example

```jsx
function UserTodos() {
  const { data: sessionData } = useSession();
  const result = useQuery({
    queryKey: ['userTodos', sessionData?.user.id],
    queryFn: fetchUserTodos,
  })
  // ...
}
```

It seems that in `tRPC`, `queryKey` is managed by the framework, so I'm wondering if it is possible to do the same.

Thanks

**Comment by Max0u** [comment]
Did you find any solutions ?
